### PR TITLE
[OREE-2061] show configuration in outreach context

### PR DESCRIPTION
**Pull request by sheff3rd**
# [OREE-2061](https://outreach-io.atlassian.net/browse/OREE-2061)

## Description

Applicaiton configuraton is inaccessible in outreach context. This fixes the issue

This line is removed because it messed with the expected output of the outreach context. It expected ConfigurationValue, but got ConfigurationItem instead
```javascript
if (app.configuration) {
  application.configuration = app.configuration.map((item) => Object.assign(new ConfigurationItem(), item));
}
```

[OREE-2061]: https://outreach-io.atlassian.net/browse/OREE-2061?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -650,6 +650,7 @@ export class ExtensibilitySdk {
     outreachContext.locale = runtime.locale;
     outreachContext.theme = runtime.theme;
     outreachContext.userIdentifier = runtime.userIdentifier;
+    outreachContext.config = runtime.configuration;
```
Comment (honeydripper) [inline src/index.ts:653]: Can we add a test for config item on the Context object?

Comment (honeydripper) [inline src/index.ts:653]: what does runtime configuration contain? It could be some private keys. outreachContext.config should only contain values that are set in the app configuration in the dev portal.

Furthermore, `config` is an object and thus this will get assigned by reference. Where as locale, theme and userIdenfier are scalar and the value will be copied. We should definitely copy the object here.

**Review by palootcenas-outreach — APPROVED** [review]

File: src/legacy/ManifestTranslator.ts [hunk 2]
```diff
@@ -309,10 +308,6 @@ export class ManifestTranslator {
       application.api = Object.assign(new ManifestApi(), app.api);
     }
 
-    if (app.configuration) {
-      application.configuration = app.configuration.map((item) => Object.assign(new ConfigurationItem(), item));
```
Comment (honeydripper) [inline src/legacy/ManifestTranslator.ts:313]: this looks like an unrelated change. It has been there since 2022 there are apps using config feature we must make sure we don't break anything.

**Review by honeydripper — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (sheff3rd) [reply src/index.ts]: this is how it'd look like. I've set some values in app configuration in devportal, than filled out some values when installing an app:

image

**Review by honeydripper — APPROVED** [review]
